test/controllers/api/v2/repositories_controller_test.rb

```diff
@@ -476,6 +476,33 @@ def test_sync
       assert_response :success
     end
 
+    def test_sync_with_url_override
+      assert_async_task ::Actions::Katello::Repository::Sync do |repo, pulp_task_id, source_url|
+        repo.id.must_equal(@repository.id)
+        pulp_task_id.must_equal(nil)
+        source_url.must_equal('file:///tmp/')
+      end
+      post :sync, :id => @repository.id, :source_url => 'file:///tmp/'
+      assert_response :success
+    end
+
+    def test_sync_with_bad_url_override
+      post :sync, :id => @repository.id, :source_url => 'file:|||tmp/'
+      assert_response 400
+    end
+
+    def test_sync_no_feed_urls
+      repo = katello_repositories(:feedless_fedora_17_x86_64)
+      post :sync, :id => repo.id
+      assert_response 500
```

Comment:
I thought it should be a 400, but it looks like it currently 500's and i didn't want to alter an API. It is from the code below (source_url check was added by me):

```ruby
          if repo.url.blank? && source_url.blank?
             fail _("Unable to sync %s. This repository does not have a feed url.")
           end
```